tests/ovirt/basic_vm_import_negative_test.go

```diff
@@ -134,9 +138,44 @@ var _ = Describe("VM import", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(created).To(HaveValidationFailure(f, string(v2vv1alpha1.ResourceMappingNotFound)))
 	})
+
+	It("should be stuck when source VM does not shut down", func() {
+		vmID := vms.BasicVmID
+		vmi := utils.VirtualMachineImportCr(vmID, namespace, secret.Name, f.NsPrefix, true)
+
+		vmXML := f.LoadTemplate("vms/status-template.xml", map[string]string{"@VMID": vmID, "@VMSTATUS": "up"})
+		actionXML := "<action/>"
+		builder := test.prepareVmResourcesStub(vmID).
+			Stub(sapi.Stubbing{
+				Path:   "/ovirt-engine/api/vms/" + vmID + "/shutdown",
+				Method: "POST",
+				Responses: []sapi.RepeatedResponse{
+					{
+						ResponseBody: &actionXML,
+						ResponseCode: 200,
+					},
+				},
+			}).
+			StubGet("/ovirt-engine/api/vms/"+vmID, &vmXML)
+		test.recordStubbing(builder)
+
+		created, err := f.VMImportClient.V2vV1alpha1().VirtualMachineImports(namespace).Create(&vmi)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		Consistently(func() (*v2vv1alpha1.VirtualMachineImportCondition, error) {
+			retrieved, err := f.VMImportClient.V2vV1alpha1().VirtualMachineImports(f.Namespace.Name).Get(created.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+			condition := conditions.FindConditionOfType(retrieved.Status.Conditions, v2vv1alpha1.Processing)
+			return condition, nil
+
+		}, 5*time.Minute, time.Minute).Should(BeNil())
```

Comment:
Should we also check that after the timeout, we get the error?

Reply:
What error do you have in mind? The timeout here only specifies how long the assertion should be kept, it does not return error after the timeout. In this case it's 5 minutes, which is as long as the ovirt client waits for the VM to be down in one reconciliation and that long I wanted to make sure that it doesn't start processing.

Reply:
I mean this one: https://github.com/kubevirt/vm-import-operator/blob/master/pkg/providers/ovirt/client/ovirt-client.go#L139

When the timeout (5min) is out, the stopVm method should return error.

Reply:
It does but on high level it is translated only to reconciliation error and CR re-queuing (https://github.com/kubevirt/vm-import-operator/blob/12de4cca49c113ce028fcb84934a64541009922e/pkg/controller/virtualmachineimport/virtualmachineimport_controller.go#L239); nothing we can detect except for verifying it doesn't progress beyond mapping rules verification.